src/Front/Ast.hs

```diff
@@ -7,11 +7,29 @@ import           Common.Pretty
 
 import           Data.List                      ( intersperse )
 
--- | A complete program: a list of declarations
-newtype Program = Program [Definition]
+-- | A complete program: a list of top-level definitions.
+newtype Program = Program [TopDef]
   deriving (Eq, Show)
 
--- | A value definition
+-- | A top-level definition.
+data TopDef
+  = TopDef Definition     -- ^ Bind a (data) value to a variable
+  | TopType TypeDef       -- ^ Declare a type
+  deriving (Eq, Show)
+
+-- | A type definition.
+data TypeDef = TypeDef
+  { typeName     :: Identifier      -- ^ The name of the type, e.g., @Option@
+  , typeParams   :: [Identifier]    -- ^ List of type parameters, e.g., @a@
+  , typeVariants :: [TypeVariant]   -- ^ List of variants/data constructors
```

Comment:
Can we add some examples for `typeVariants` too? Maybe `None` and `Some`, in the context of `Option`?

Reply:
will do